### DISCUSSION TITLE
Expected behavior for navigation with search param changes

### DISCUSSION THREAD

**Discussion by akspex** [TanStack/router#2879] [General]
Hey there,

what is the expected behavior when clicking on a `Link` that targets the same route but with a changed search param? 

Example:
- you have a route called /my-list
- that route renders some list with many items thus it is partitioned in "pages"
- there are `Link` components for navigating to another page, e.g. when on page 1, there could be a link to the next page 2 which should navigate to `/my-list?page=2`
- what is the expected behavior when clicking on that link?
- are the routes `/my-list?page=1` and `/my-list?page=2` treated as a different page?
- I'd assume they are different and thus should set the scroll position back to the top of the page when navigating to them via a `Link`

Currently though a search param change like described above retains the current scroll position. That means for example that a click on a pagination button at the bottom of a long list changes the search parameter in the url and gets picked up by e.g. loaders but the page stays scrolled down to the pagination button.

Is my basic assumption wrong, that `/my-list?page=1` and `/my-list?page=2` are treated as separate pages and thus reset the scroll position?

Is there a way to get to my expected behavior of having the scroll position reset when clicking on such a link? For some reason `resetScroll={true}` does not seem to work at all in our case.

Maybe helpful for this discussion: I created a minimal example at https://stackblitz.com/edit/github-3fv17a?file=src%2Fmain.tsx where you can see a similar example to what I described. A click on a link does change the search params and re-render (parts of) the page but the scroll position does not reset to top.

**Comment by schiller-manuel** [comment]
there was a bug that is now fixed
here is your updated example: https://stackblitz.com/edit/github-3fv17a-e9ekcj?file=src%2Fmain.tsx,package.json

**Reply by akspex** [reply]
Thanks for the fast reply! I was able to verify today that indeed the scroll position is behaving as expected, changes in search params are indeed behaving like a route change. Awesome, thanks for your work!

**Comment by akspex** [comment]
This has been answered by the fact that there was a fix which resulted in now treating search param changes as url changes.
